### ISSUE TITLE
Use GraalVM JDK shouldn't be mandatory for dockerBuildNative

### ISSUE THREAD

**Issue by ilopmar** [micronaut-projects/micronaut-gradle-plugin#49]
If a user wants to create a native image inside a docker container it shouldn't be mandatory to use GraalVM JDK for running the command, because the native-image is generated inside the container.

```
$ java -version            
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-b01)
OpenJDK 64-Bit Server VM (build 25.265-b01, mixed mode)

$ ./gradlew dockerBuildNative
> Task :complete:compileJava FROM-CACHE
> Task :complete:processResources UP-TO-DATE
> Task :complete:classes UP-TO-DATE
> Task :complete:runnerJar
> Task :complete:buildLayers
> Task :complete:dockerfileNative FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':complete:dockerfileNative'.
> A GraalVM SDK is required to build native images
```

With GraalVM JDK:

```
$ ./gradlew dockerBuildNative
> Task :complete:compileJava FROM-CACHE
> Task :complete:processResources
> Task :complete:classes
> Task :complete:runnerJar
> Task :complete:buildLayers

> Task :complete:dockerfileNative
Dockerfile written to: /home/ivan/workspaces/micronaut/micronaut-guides/micronaut-creating-first-graal-app/complete/build/docker/DockerfileNative

> Task :complete:dockerBuildNative
Building image using context '/home/ivan/workspaces/micronaut/micronaut-guides/micronaut-creating-first-graal-app/complete'.
Using Dockerfile '/home/ivan/workspaces/micronaut/micronaut-guides/micronaut-creating-first-graal-app/complete/build/docker/DockerfileNative'
Using images 'complete'.
...
...
```

I understand that for `nativeImage` task is mandatory to use GraalVM JDK and that both tasks use the `dockerfileNative` task but maybe the check could moved and only executed in former instead of the latter.

### More info

Plugin version: 1.0.3

**Comment by graemerocher** [comment]
I disagree, we would have to add all sorts of checks to check whether the use built on an unsupported JDK version such as 12/13/14/15 to the plugin.

The alternative would be to build the project with docker but that means dealing with how to cache dependencies inside the docker build and all sorts of other problems.

**Comment by ilopmar** [comment]
I understand this, but from a user point of view we have now less functionality than before. 
Without the plugin the users only needed to generate the fatjar and run the docker build script. They didn't need to install GraalVM. Now they do.

I agree that if user builds the fatjar locally with JDK 14, then the native-image generation would fail.

**Comment by graemerocher** [comment]
IMO this is a better user experience ensuring that they build with GraalVM rather than leaving it up to chance whether they have a correct version of Java or not that is compatible with native image.

I am ok with making the behaviour configurable but ultimately the previous behaviour of allowing the user to build with an unsupported JDK then failing later was worse.

**Comment by graemerocher** [comment]
I think what I will do is limit the JDK to a maximum of 11 when running `./gradlew dockerBuildNative`

**Comment by alvarosanchez** [comment]
If the native image is built inside a container, I still don't get why having GraalVM locally matters

**Comment by graemerocher** [comment]
because the class files are built locally and copied into the container so if you build with a JDK that is incompatible with native image (JDK 12+) then native image will fail

**Comment by alvarosanchez** [comment]
I understand, but still doesn't make sense. Doesn't make sense to require them build the JAR locally with GraalVM, then use native-image inside a container to build the binary.

If I were to have GraalVM locally, I would just use `nativeImage`.

The point of `dockerBuildNative`, IMO, is to be able to build a native image regardless of what I use locally. You could craete a multi-stage Docker image with a build stage that has `FROM oracle/graalvm-ce:20.2.0-java8` or `FROM oracle/graalvm-ce:20.2.0-java11` and then `./gradlew assemble` it.

**Comment by graemerocher** [comment]
That is a bad user experience IMO as you have to redownload the gradle dependencies and wrapper on every build unless you figure out some complex way to cache the gradle cache in a docker layer

**Comment by ilopmar** [comment]
What if we only check, as @graemerocher already mentioned that they are using a maximum of JDK 11 (no matter which JDK-vendor)? 
This way the users can build the application with a non-GraalVM JDK and if "it's valid" (JDK 8 or 11), then we allow to create the native image using it.

**Comment by graemerocher** [comment]
That seems reasonable to me

**Comment by alvarosanchez** [comment]
Yes, it is possible, and yes, this was done if you created an application with the `graalvm` and `aws-custom-runtime` features.

**Comment by graemerocher** [comment]
The downside of this approach is you have to replicate the gradle cache and wrapper in a docker layer which is wasteful imo. I think it would be better just to have a check that the java source target version is set to 11 or below

**Comment by alvarosanchez** [comment]
The docker layer will be cached, so it will only have to download wrapper/dependencies once.

And it shouldn't affect the final image size since that build stage would be removed.

**Comment by graemerocher** [comment]
Fixed by https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/58
